include/RAJA/policy/hip/kernel/For.hpp

```diff
@@ -623,35 +489,29 @@ struct HipStatementExecutor<
   using diff_t = segment_diff_type<ArgumentId, Data>;
 
 
-  static
-  inline RAJA_DEVICE void exec(Data &data, bool thread_active)
+  static inline RAJA_DEVICE
+  void exec(Data &data, bool thread_active)
   {
     diff_t len = segment_length<ArgumentId>(data);
-    diff_t i = get_hip_dim<BlockDim>(dim3(blockIdx.x,blockIdx.y,blockIdx.z));
-
-    if (i < len) {
+    diff_t i = Indexer::template index<diff_t>();
 
-      // Assign the x thread to the argument
-      data.template assign_offset<ArgumentId>(i);
+    // Assign the index to the argument
+    data.template assign_offset<ArgumentId>(i);
 
-      // execute enclosed statements
-      enclosed_stmts_t::exec(data, thread_active);
-    }
+    // execute enclosed statements
+    enclosed_stmts_t::exec(data, thread_active && (i < len));
```

Comment:
I always set the thread_active instead of using a conditional in the new version. This is probably ok but may cause some blocks to do more busy work than before. This is done to keep the threads in a warp from diverging when mapping to threads to allow things like block and warp synchronizes.

Reply:
For example https://github.com/LLNL/RAJA/blob/c427cf604fda80689118dbd79e12afcb77445f83/test/old-tests/unit/test-kernel.cpp#L2329

Reply:
The above use case seems to be the only one we test where we synchronize across some parallel resources while inside a mapping of said parallel resources. So we can't remove support for this use case, but maybe long term we could find another way to support it so we can avoid this mismatch. Alternatively we could add a special policy to support this use case.

Reply:
Related to the active thread concept I have changed the thread loop policies to be non-synchronize-able by default. I don't know if anyone actually relied on that, if not maybe we could remove the synchronize-able concept entirely. I did this somewhat against my will because launch only has a non-synchronize-able implementations. So the same polices work with launch and kernel, but the semantics of thread loop policies like cuda_thread_x_loop changes in kernel.